imagebuildah/build.go

```diff
@@ -1225,3 +1241,52 @@ func BuildDockerfiles(ctx context.Context, store storage.Store, options BuildOpt
 	stages := imagebuilder.NewStages(mainNode, b)
 	return exec.Build(ctx, stages)
 }
+
+// preprocessDockerfileContents runs CPP(1) in preprocess-only mode on the input
+// dockerfile content and will use ctxDir as the base include path.
+//
+// Note: we cannot use cmd.StdoutPipe() as cmd.Wait() closes it.
+func preprocessDockerfileContents(r io.ReadCloser, ctxDir string) (*io.ReadCloser, error) {
```

Comment:
Add err error;

Reply:
This will not work correctly unless you add.

func preprocessDockerfileContents(r io.ReadCloser, ctxDir string) (*io.ReadCloser, err error) {
                                                                                                                                   ^^^

Reply:
Mixed named return values don't compile. I'll figure out what to do with the ReadCloser.